src/streaming-ingestion-redshift-stack.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import {
+  CfnCondition,
+  CfnStack,
+  Fn,
+  Stack,
+  StackProps,
+} from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { SolutionInfo } from './common/solution-info';
+import {
+  createStackParameters, StreamingIngestionRedshiftStackProps,
+} from './streaming-ingestion/parameter';
+import { StreamingIngestionRedshiftStack } from './streaming-ingestion/streaming-ingestion-redshift';
+import { REDSHIFT_MODE } from '../src/common/model';
+
+export class StreamingIngestionRedshiftMainStack extends Stack {
+  public readonly nestedStacks: {
+    readonly redshiftServerlessStack: StreamingIngestionRedshiftStack;
+    readonly redshiftProvisionedStack: StreamingIngestionRedshiftStack;
+  };
+  constructor(
+    scope: Construct,
+    id: string,
+    props?: StackProps,
+  ) {
+    super(scope, id, props);
+
+    const featureName = 'StreamingIngestion';
+    this.templateOptions.description = `(${SolutionInfo.SOLUTION_ID}-sir) ${SolutionInfo.SOLUTION_NAME} - ${featureName} ${SolutionInfo.SOLUTION_VERSION_DETAIL}`;
+
+    const p = createStackParameters(this);
+    this.templateOptions.metadata = p.metadata;
+
+    this.nestedStacks = createStreamingIngestionRedshiftStack(this, p.params);
+  }
+}
+
+export function createStreamingIngestionRedshiftStack(
+  scope: Construct,
+  props: StreamingIngestionRedshiftStackProps,
+) {
+  const nestStackProps = {
+    projectId: props.projectId,
+    appIds: props.appIds,
+    streamingIngestionProps: {
+      streamingIngestionRoleArn: props.streamingIngestionConfiguration.streamingIngestionRoleArn,
+      kinesisDataStreamProps: {
+        kinesisDataStreamNames: props.streamingIngestionConfiguration.kinesis?.kinesisDataStreamNames,
+      },
+    },
+  };
+
+  const redshiftModeStr = props.redshift.mode;
+
+  const isExistingRedshiftServerless = new CfnCondition(
+    scope,
+    'existingRedshiftServerless',
+    {
+      expression:
+        Fn.conditionEquals(redshiftModeStr, REDSHIFT_MODE.SERVERLESS),
+    },
+  );
+  const isRedshiftProvisioned = new CfnCondition(
+    scope,
+    'redshiftProvisioned',
+    {
+      expression:
+        Fn.conditionEquals(redshiftModeStr, REDSHIFT_MODE.PROVISIONED),
+    },
+  );
+
+  const redshiftExistingServerlessStack = new StreamingIngestionRedshiftStack(
```

Comment:
CloudFormation can only distinguish between `RedshiftMode` through `CfnCondition`, otherwise, it is impossible to determine whether to set serverless or provisioned parameters.